gemrb/plugins/SDLAudio/SDLAudio.cpp

```diff
@@ -340,32 +345,33 @@ Holder<SoundHandle> SDLAudio::Play(const char* ResRef, unsigned int channel,
 
 int SDLAudio::CreateStream(Holder<SoundMgr> newMusic)
 {
+	std::lock_guard<std::recursive_mutex> l(MusicMutex);
```

Comment:
Adding this lock implies `MusicReader` needs to be protected by the lock, yet we don't do that everywhere.

Reply:
But we're doing that in `music_callback`

Reply:
right, but we need to hold the lock _every_ time we touch it and we are not doing that. So I'm not suggesting a change here, I'm suggesting that we add a lock to the other locations.